### PR TITLE
Hw 4

### PR DESCRIPTION
**Pull request by VodopianovRoman**

### CODE REVIEW

File: hw_decorator.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+#     return int_n ** int_n
+#
+# print(multi_bulti(5))
+
+
+
+# def decorator_fib(func):
+#     cache = {}
+#     ncalls = 0
+#     ccalls = 0
```
Comment (katevoskresenska) [inline hw_decorator.py:97]: wraps.ncalls and wraps.ccalls are more preferable than nonlocal variables as they can be re-used

**Comment by katevoskresenska** [comment]
It would be nice to add some documentation strings to all the functions.

File: hw_decorator.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+
+# def decorator_remainder_of_division(func):
+#     @functools.wraps(func)
+#     def wraper(*args, **kwargs):
+#         res = 1
+#         for i in args:
+#             res *= i
+#         res_remainder = 100 % res
+#         if res_remainder == 0:
```
Comment (katevoskresenska) [inline hw_decorator.py:16]: if not remainder:

File: hw_3.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
+# 1) Write a function that emulates the game "rock, scissors, paper"
```
Comment (mmykhailenko) [inline hw_3.py:1]: this file should not be here in this PR

**Review by mmykhailenko — APPROVED** [review]

**Pull request merged**